src/gr_generic/generic.c

```diff
@@ -117,6 +117,49 @@ int gr_generic_randtest_not_zero(gr_ptr x, flint_rand_t state, gr_ctx_t ctx)
     return GR_UNABLE;
 }
 
+int gr_generic_randtest_invertible(gr_ptr x, flint_rand_t state, gr_ctx_t ctx)
+{
+    slong i;
+    truth_t is_invertible;
+    int status = GR_SUCCESS;
+
+    for (i = 0; i < 5; i++)
+    {
+        status |= gr_randtest(x, state, ctx);
+
+        is_invertible = gr_is_invertible(x, ctx);
+        if (is_invertible == T_TRUE)
+            return GR_SUCCESS;
+    }
+
+    for (i = 0; i < 5; i++)
+    {
+        status |= gr_randtest_small(x, state, ctx);
+
+        is_invertible = gr_is_invertible(x, ctx);
+        if (is_invertible == T_TRUE)
+            return GR_SUCCESS;
+    }
+
+    if (n_randint(state, 2))
```

Comment:
At this point we can rather do ``return n_randint(state, 2) ? gr_one(x, ctx) : gr_neg_one(x, ctx)`` as 1 and -1 are always invertible.